.circleci/config.yml

```diff
@@ -0,0 +1,190 @@
+version: 2.1
+
+parameters:
+  snapshot:
+    type: boolean
+    default: true
+  release:
+    type: boolean
+    default: false
+
+orbs:
+  aws-white-list-circleci-ip: configure/aws-white-list-circleci-ip@1.0.1
+
+executors:
+  docker-executor:
+    docker:
+      - image: 455456581940.dkr.ecr.eu-west-1.amazonaws.com/circleci-build-images:corretto-8u275
+        aws_auth:
+          aws_access_key_id: $AWS_ACCESS_KEY_ID
+          aws_secret_access_key: $AWS_SECRET_ACCESS_KEY
+
+jobs:
+  build-and-deploy:
+    executor: docker-executor
+    steps:
+      - checkout
+      - get-maven-settings-file
+      - restore-cache
+      - whitelist-add
+      - run:
+          name: "Setting Maven version"
+          command: |
+            VERSION=$(mvn -s gresham-nexus-settings/ctc.plugins.settings.xml -Dexec.executable='echo' -Dexec.args='${project.version}' --non-recursive exec:exec -q)
+            if [[ "${CIRCLE_BRANCH}" != "1.3.x" && "${VERSION}" == *-SNAPSHOT ]]; then
+              mvn -s gresham-nexus-settings/ctc.plugins.settings.xml versions:set -DnewVersion=${CIRCLE_BRANCH}.GRESHAM-SNAPSHOT -B
+            fi
+      - run:
+          name: "Running Maven build and deploy"
+          command: |
+            MASTER_BRANCH=1.3.x
```

Comment:
Can this be defined earlier on? Only because in the bump version a few lines above you have a hardcoded 1.3.x string.

Reply:
CCI says no. Each run command runs in its own shell, so any env var set before won't persist. I'll compromise...